### ISSUE TITLE
"Trailing Whitespaces" rule has false positives with comment blocks

### ISSUE THREAD

**Issue by steventrouble** [pinterest/ktlint#932]
Example failure:

```
data class Foo (
  /** The user's bar object. */
  val bar: Bar
)
```

Fails with "No trailing space(s)", because it thinks the first two characters of the comment are the entire line. This should not fail, as it makes it impossible to document data class parameters using the inline format supported by KDoc.

**Comment by romtsn** [comment]
What version of ktlint are you using? I couldn't reproduce it. Are you sure it is the `no-trailing-spaces` rule? You can run ktlint with `--verbose` to see the rule id.

**Comment by romtsn** [comment]
Feel free to re-open with a reproducible sample

**Comment by rnett** [comment]
I'm seeing the same thing with `0.41.0`:
```
ktlint:
   [ktlint] WARNING: An illegal reflective access operation has occurred
   [ktlint] WARNING: Illegal reflective access by org.jetbrains.kotlin.com.intellij.util.ReflectionUtil (file:/home/rnett/.m2/repository/org/jetbrains/kotlin/kotlin-compiler-embeddable/1.4.31/kotlin-compiler-embeddable-1.4.31.jar) to method java.util.ResourceBundle.setParent(java.util.ResourceBundle)
   [ktlint] WARNING: Please consider reporting this to the maintainers of org.jetbrains.kotlin.com.intellij.util.ReflectionUtil
   [ktlint] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
   [ktlint] WARNING: All illegal access operations will be denied in a future release
   [ktlint] /windows/Users/jimne/Desktop/OtherStuff/tensorflow_java/tensorflow-core-kotlin/tensorflow-core-kotlin-api/src/gen/annotations/org/tensorflow/op/kotlin/IoOps.kt:1588:33: Trailing space(s) (no-trailing-spaces)
```

Where the relevant line is [here](https://gist.github.com/rnett/3579086d155d7142f3ff9aa012dce427#file-ioops-kt-L1588).

**Comment by rnett** [comment]
Notably, this causes `-F` to fail, but if you run `-F` again immediately afterwards, it succeeds in fixing it.  The file linked above is from after the first `-F` run, obtained by using `ktlint` without `-F`.

**Comment by Tapchicoma** [comment]
@rnett please open a new issue.
